india_compliance/gst_india/overrides/transaction.py

```diff
@@ -1281,10 +1318,30 @@ def before_validate(doc, method=None):
 def update_gst_details(doc, method=None):
     ItemGSTTreatment().set(doc)
     if doc.doctype in DOCTYPES_WITH_GST_DETAIL:
+        validate_gst_accounts_for_taxable_invoice(doc)
         ItemGSTDetails().update(doc)
         validate_non_taxable_items(doc)
 
 
+def validate_gst_accounts_for_taxable_invoice(doc):
+    if frappe.flags.in_test:
+        return
+
+    is_taxable_invoice = any(
+        item.gst_treatment == "Taxable" and item.taxable_value != 0
+        for item in doc.items
+    )
+
+    if not is_taxable_invoice:
+        return
+
+    gst_accounts = get_all_gst_accounts(doc.company)
+    has_gst_accounts = any(tax.account_head in gst_accounts for tax in doc.taxes)
+
+    if not has_gst_accounts:
+        frappe.throw(_("No GST Accounts has been charged for GST Taxable Invoice"))
```

Comment:
"No GST is charged on this transaction. Do you wish to continue without GST?

Select Exclude from GST and continue.
"